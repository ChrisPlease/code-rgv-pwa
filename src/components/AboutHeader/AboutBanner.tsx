/**
 * About Title Component
 */
import * as React from "react";
import { v4 } from "uuid";
import { map } from "ramda";
import { HeaderOne } from "../Headers";
import { AboutBackgroundSlice } from "./AboutBackgroundSlice";
import { tap } from "ramda";

const titleStyle = {
  color: "#FFAA00",
  textAlign: "center"
};

const h1 = HeaderOne.contramap((t: { headerText: string }) => ({
  headerText: t.headerText,
  style: titleStyle
}));

export const AboutBanner = AboutBackgroundSlice.contramap(
  (t: { [key: string]: string }) => ({
    title: h1.fold(t)
  })
);
