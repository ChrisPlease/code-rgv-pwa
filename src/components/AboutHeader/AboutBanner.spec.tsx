/**
 * About Banner test
 */
import { shallow } from "enzyme";
import * as renderer from "react-test-renderer";
import * as React from "react";
import { AboutBanner } from "./AboutBanner";

test("About Banner", () => {
  const component = renderer.create(AboutBanner.fold({ text: "test" }));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
