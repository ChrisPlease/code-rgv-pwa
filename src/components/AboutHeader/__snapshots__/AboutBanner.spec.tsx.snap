// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Banner 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#00b3b3",
      "height": "200px",
      "marginBottom": "30px",
      "paddingTop": "70px",
    }
  }
>
  <h1
    className="title is-1"
    style={
      Object {
        "color": "#FFAA00",
        "textAlign": "center",
      }
    }
  >
    test
  </h1>
</div>
`;
